static/js/jira-server-url.js

```diff
@@ -68,18 +68,50 @@ const requestFailed = () => {
 	$("#gheServerBtnSpinner").hide();
 };
 
+const gheServerUrlErrors = {
+	errorCode: {
+		GHE_ERROR_1: {
+			title: "Invalid URL",
+			message: "That URL doesn't look right. Please check and try again.",
+		},
+		GHE_ERROR_2: {
+			title: "We couldn't verify this URL",
+			message: "Please make sure you've entered the correct URL and check that you've properly configured the hole in your firewall.",
+		},
+		GHE_ERROR_3: {
+			title: "Request failed",
+			message: "We weren't able to complete your request. Please try again."
+		},
+		GHE_ERROR_4: {
+			title: "Something went wrong",
+			message: "We ran into a hiccup while verifying your details. Please try again later."
+		}
+	}
+};
+
+const handleGheUrlRequestErrors = (err) => {
+	requestFailed();
+	$(".jiraServerUrl__validationError").show();
+	$(".errorMessageBox__title").append(title);
+	$(".errorMessageBox__message").append(message);
+	title === gheServerUrlErrors.errorCode.GHE_ERROR_2.title && $(".errorMessageBox__link").show();
+}
+
 const verifyGitHubServerUrl = (gheServerURL, installationId) => {
-	const csrf = document.getElementById('_csrf').value;
+	const csrf = document.getElementById("_csrf").value
 
-	window.AP.context.getToken(function(token) {
-		$.post("/jira/server-url", {
+	AP.context.getToken(function(token) {
+		$.ajax({
```

Comment:
Had to change this from the initial implementation as the 502 error seemed to be getting swallowed. Everything works as expected this way.

Reply:
all the github error handling should be done in the success function, the error function is just for our own server issues, which I'm not sure how to handle in this case...